aten/src/ATen/native/mobile/cpu/internal/Linear.cpp

```diff
@@ -0,0 +1,193 @@
+      0u,                                                               // flags
+      &linear_op);                                                      // operator
+
+  TORCH_INTERNAL_ASSERT(
+      xnn_status_success == create_status,
+      "xnn_create_fully_connected_nc_f32 failed!");
+
+  return Context{
+    Operator(linear_op),
+    {
+      weight.size(Layout::Filter::Output),
+    }
+  };
+}
+
+bool usable(const Tensor& input) {
+         // Input
+  return (2 <= input.ndimension()) &&
+         (c10::DeviceType::CPU == input.device().type()) &&
+         (kFloat == input.scalar_type());
+}
+
+Tensor run(
+    const Context& context,
+    const Tensor& input) {
+  using namespace internal;
+
+  TORCH_INTERNAL_ASSERT(
+      usable(input),
+      "mobile::cpu::linear not usable!");
+
+  const IntArrayRef input_size = input.sizes();
+  std::vector<int64_t> output_size(input_size.cbegin(), input_size.cend());
+  output_size.back() = context.output.channels;
+
+  Tensor output = empty(
+      output_size,
+      input.options(),
+      input.suggest_memory_format());
+
+  const xnn_status setup_status = xnn_setup_fully_connected_nc_f32(
+      context.linear_op.get(),                    // operator
+      Layout::ActivationND::Batch(input.sizes()), // Batch,
+      input.data_ptr<float>(),                    // input
+      output.data_ptr<float>(),                   // output
+      nullptr);                                   // threadpool
+
+  TORCH_INTERNAL_ASSERT(
+      xnn_status_success == setup_status,
+      "xnn_setup_fully_connected_nc_f32 failed!");
+
+  const xnn_status run_status = xnn_run_operator(
+      context.linear_op.get(),  // operator
+      nullptr);                 // threadpool
+
+  TORCH_INTERNAL_ASSERT(
+      xnn_status_success == run_status,
+      "xnn_run_operator failed!");
+
+  return output;
+}
+
+Tensor create_and_run(
+    const Tensor& input,
+    const Tensor& weight,
+    const Tensor& bias,
+    const Scalar output_min,
+    const Scalar output_max) {
+  return run(
+      create(
+          weight,
+          bias,
+          output_min,
+          output_max),
+      input);
+}
+
+} // namespace
+} // namespace linear
+} // namespace internal
+
+bool use_linear(
+    const Tensor& input,
+    const Tensor& weight,
+    const Tensor& bias) {
+  return internal::linear::available(
+            weight,
+            bias,
+            internal::linear::Context::kMin,
+            internal::linear::Context::kMax) &&
+         internal::linear::usable(input);
+}
+
+Tensor linear(
+    const Tensor& input,
+    const Tensor& weight,
+    const Tensor& bias) {
+  return internal::linear::create_and_run(
+      input,
+      weight,
+      bias,
+      internal::linear::Context::kMin,
+      internal::linear::Context::kMax);
+}
+
+} // namespace cpu
+} // namespace mobile
+} // namespace native
+} // namespace at
+
+namespace caffe2 {
+
+CAFFE_KNOWN_TYPE(at::native::mobile::cpu::internal::linear::Context);
```

Comment:
What's going on here?

Reply:
https://github.com/pytorch/pytorch/blob/64323ae17789adda57f3b0fd4b0739e133b3a38d/aten/src/ATen/cpp_custom_type_hack.h#L8

Reply:
In hindsight, we don't need this anymore now that I removed the c10 custom op registration stuff.